File: targetgenerator/weather.js
var Wind = function(direction, velocity, gust) {
  this.direction = direction || 0.0;
  this.velocity = velocity || 0.0;
  this.gust = gust || 0.0;
};

var Station = function(identifier, position, wind) {
  this.identifier = identifier || '';
  this.position = position || null;
  this.wind = wind || null;
};

var Situation = function() {
  this.stations = [];
};

Situation.prototype.windAtPosition = function(position) {
  var wind = new Wind();
  var closest = Number.MAX_VALUE;
  this.stations.forEach(function(station) {
    var distance = station.position.distanceTo(position);
    if (distance < closest)
    {
      closest = distance;
      wind = station.wind;
    }
  }.bind(this));
  return wind;
};

module.exports = {
  Wind: Wind,
  Station: Station,
  Situation: Situation
};

File: targetgenerator/weather.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TargetGenerator
{
    class Wind
    {
        public double direction { get; set; }
        public double velocity { get; set; }
        public double gust { get; set; }
    }

    class WeatherStation
    {
        public string identifier { get; set; }
        public Position position { get; set; }
        public Wind wind { get; set; }
    }

    class WeatherSituation
    {
        public WeatherStation[] stations { get; set; }

        public Wind windAtPosition(Position position)
        {
            Wind wind = new Wind();
            double closest = Double.MaxValue;
            foreach (WeatherStation station in stations)
            {
                double distance = station.position.distanceTo(position);
                if (distance < closest)
                {
                    closest = distance;
                    wind = station.wind;
                }
            }
            return wind;
        }
    }
}